### DISCUSSION TITLE
How does v1.3.4 work for y'all?

### DISCUSSION THREAD

**Discussion by bazmonk** [bazmonk/digigurdy-baz#69] [Q&A]
@Gubbledenut and @lune36400, my treasured bug testers!

The rectifier and reversible crank idea have made smoothness vs. sensitivity more important with the crank.  It didn't used to matter so much before but now that staccato is possible... so the recent versions of the code have made a lot of strides (I think) in that regard...

* Grounding my motor (and finding the "right" diode for the job) is enabling really good crank response.
* I'm going with faster sampling now and larger "spin" values in order to be able to fine-tune them more.
* I'm using a rolling average to smooth the voltage from cycle to cycle.
* I noticed that sometimes (randomly) the "resting" voltage reading off the pin will be upwards of 4-5 above 0.  So I got the idea to take an average of the resting voltage (which we happen to already gather during the crank detection routine!) and subtract that from the voltage readings to "normalize" them.  Because the choice of `V_THRESHOLD` has to be higher than this resting voltage to prevent phantom sounds, normalizing it like this allows me to lower it, which has a HUGE effect on how sensitive-and-smooth is possible with the other settings.  
  * If either of you look at the Serial Monitor output on these recent versions, I print out the "raw" sampled voltage, the rolling average, and the normalized version of that which is actually used.  You can see how it effects it all.

The bottom line is that I think I've got it fine-tuned really, really good for my crank, but now that I'm squeezing every bit of performance out of it, I'm not sure how it behaves with other people's cranks.  Before it was easy to choose very-smooth settings and trust it would work for most people.

So I'm hoping (no rush) that the two of you could try out v1.3.4 and let me know:

* How low of a`V_THRESHOLD` can you use without getting phantom sounds now that I'm normalizing it?
* Is your crank smooth enough at slow speed?  If not, lower `SPIN_DECAY` only (leave the rest alone for this "experiment", please).  How low do you have to set it until it's "smooth enough" for you?

I know the settings I like, but I'm hoping to combine that with the settings you guys use and use them as the "default" settings I keep on the repository for your average person looking for "safe" settings.

I do plan on making these two values adjustable on-screen soon.  They seem to be enough to adjust the overall responsiveness without messing with the other settings at all.

Thanks if you decide to help!

**Comment by Gubbledenut** [comment]
first the config.h file is a dream come true ..thank you......
I still have my 4007 diode rectifier in so need to spin pretty fast to not jitter...but I do see how things are smoother..and hopefully my 60p diodes will help..and nope no 5817 in the country , I may see if any friends have spares , if the 60p don't work well enough.
So give me a few more days..so I can test properly with a better or no rectifier...

**Reply by bazmonk** [reply]
Which country is this far-off land?  I could probably get you some and mail it over if you want.

**Reply by bazmonk** [reply]
@Gubbledenut I jumped on it just in case: I've got 100 of them on their way to me, $0.25 USD each.  Not sure why they're not available in your country... there's no shortage of them here.

Anyway, these are the same Vishay ones that I'm using, the known good ones.  I'l let you know when they arrive.
